tst/Carnival/Providers/HookServiceProviderTest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Tests\Carnival\Providers;
+
+use Carnival\Hooks\HookManager;
+use Carnival\Providers\HookServiceProvider;
+use Illuminate\Contracts\Foundation\Application;
+use Mockery;
+use Tests\Carnival\CarnivalTestCase;
+
+class HookServiceProviderTest extends CarnivalTestCase {
+
+    public function testHookServiceProviderCanRegisterHookLibraryFacade() {
+        $application = Mockery::mock(Application::class);
+        $hookServiceProvider = new HookServiceProvider($application);
+        $application->shouldReceive('bind')->with('hooks', Mockery::on(fn ($argument) => $argument() instanceof HookManager ));
```

Comment:
Perhaps this is a nitpick, but thinking of this on principal:
Unit tests should either pass the assertion or fail the assertion. If an `$argument` comes through that is not a callable, then this test won't have failed because the assertion didn't pass, but rather because the test triggered an exception, right?

So given that principal, I'd argue for your `Mockery::on` to look like:
```
Mockery::on(fn ($argument) => is_callable($argument) && $argument() instanceof ExtensionManager)
```
Now, if $argument is not callable, we'll know that and it'll fail the assertion without triggering an exception. What do you think?

Reply:
Better safe than sorry. I think adding the is_callable check is a good idea.